appserver/concurrent/concurrent-impl/src/main/java/org/glassfish/concurrent/runtime/ContextSetupProviderImpl.java

```diff
@@ -388,7 +388,7 @@ public void reset(ContextHandle contextHandle) {
         if (handle.getSecurityContext() != null) {
             SecurityContext.setCurrent(handle.getSecurityContext());
         }
-        if (handle.getInvocation() != null) {
+        if (handle.getInvocation() != null /*&& !handle.isUseTransactionOfExecutionThread()*/) {
```

Comment:
Why adding this comment on the condition?

Reply:
removed